src/compiler.ts

```diff
@@ -2,12 +2,11 @@ import { workspace } from 'vscode';
 import * as prql from 'prql-js';
 
 export function compile(prqlString: string): string | ErrorMessage[] {
+  // create compile options from prql workspace settings
+  const compileOptions = new prql.CompileOptions();
   const target = <string>workspace.getConfiguration('prql').get('target');
-  const options: CompileOptions = { target: target };
+  compileOptions.target = `sql.${target.toLowerCase()}`;
```

Comment:
This won't work, as your dialect names are not just the same as names prql-compiler is expecting: (`PostgreSql` vs `postgres`).

I'd suggest to just change the target configuration enum to use `sql.postgres` instead of `PostgreSql`.

Reply:
Is that the only mismatch?

I pulled those names from your old Dialect enum.

Reply:
Yes, it looks like it's the same.

But regardless, using `.toLowerCase()` is not a good practice, as it introduces a hidden, non-obvious dependency between the configuration enum and the serialized names declared in prql-compiler.

Reply:
I am lowercasing it for the API I am calling it. What's not obvious about it?

I set those values to BigQuery, ClickHouse, etc. in extension settings for presentation reasons.

Reply:
Well, we have to remember that when we add dialect:

```rust
    #[strum(serialize = "yugabyte")]
    YugabyteDB
```

the configuration in the ext has to be named `Yugabyte` and not `YugabyteDB`.

It's much simpler if target configuration contains exactly the same strings as the `prql target:` header is expecting.

Reply:
FWIW I agree with having & enforcing one canonical name everywhere, but also defer to @RandomFractals as the owner here. 

We'll learn if it's the wrong decision, not worried about doing something irreversible...